InformationScripting/src/parsing/QueryParser.h

```diff
@@ -60,10 +61,10 @@ class INFORMATIONSCRIPTING_API QueryParser
 		Type typeOf(const QString& text);
 		QPair<QStringList, QList<QChar> > split(const QString& text, const QList<QChar>& splitChars);
 
-		Query* parseQuery(const QString& text);
-		QList<Query*> parseList(const QString& text);
-		Query* parseOperator(const QString& text, bool connectInput = false);
-		QList<Query*> parseOperatorPart(const QString& text);
+		Query* parseQuery(Query* parent, const QString& text);
```

Comment:
Then why not have the QueryExecutor as a field in the Query baseclass and have a setter/getter for it (it doesn't even need to be virtual), then we don't even need the TopLevelQuery. This comes at an overhead of a single pointer field in the base class which isn't that bad.?